python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -236,3 +378,48 @@ def format(self, ref):
             path = os.path.relpath(path, start="/")
 
         return path
+
+    def validateTemplate(self, entity):
+        """Compare the template against a representative entity that would
+        like to use template.
+
+        Parameters
+        ----------
+        entity : `DatasetType`, `DatasetRef`, or `StorageClass`
+            Entity to compare against template.
+
+        Raises
+        ------
+        ValidationError
+            The template is inconsistent with the supplied entity.
+
+        Notes
+        -----
+        If a `StorageClass` is supplied then validation can not be done
+        against Dimensions and it will always succeed.  If a `DatasetType`
+        or `DatasetRef` is provided without dimensions validation will
+        succeed.
+        """
+
+        # If we do not have dimensions defined then all we can do is shrug
```

Comment:
I think I should also check that one of `run` or `collection` is present in the template regardless of the supplied entity. Maybe I should do that on construction of a `FileTemplate` rather than during this validation.